### PR TITLE
feat(AP-3822): Show snippet as not detected

### PR DESCRIPTION
**Pull request by joshua-barnard-evolv**
Added new functionality that disables the Evolv snippet (in the extension UI) when no snippet is detected.
The extension also updates a lot faster than before.

### CODE REVIEW

File: src/popup.ts [hunk 1]
```diff
@@ -1,59 +1,53 @@
...
 
-const sendMessage = function (message) {
+const sendMessage = function (message: any) {
   chrome.tabs.query({
     currentWindow: true,
     active: true
   }, function (tabs) {
-    var activeTab = tabs[0];
-    chrome.tabs.sendMessage(activeTab.id, message);
+    var activeTab: chrome.tabs.Tab = tabs[0];
```
Comment (joshua-barnard-evolv) [inline src/popup.ts]: Change var to const

File: package.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "chrome-extension",
```
Comment (mattstrom) [inline package.json]:
```suggestion
  "name": "@evolv/chrome-extension",
```

File: package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  "version": "1.0.0",
+  "description": "## Getting Started --- #### **1)** In this Github project, click the green `Code` button and choose `Download ZIP`.  Save the zip file somewhere that you're not going to delete it by accident.  Once downloaded, open the zip file and you should see a directory named `evotools-main`. <img src=\"https://imgur.com/8hvUBj4.png\"/>",
+  "main": "src/background.js",
+  "type": "module",
```
Comment (mattstrom) [inline package.json:6]:
```suggestion
  "type": "module",
  "private": true,
```

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{
+  "compilerOptions": {
+    "target": "es2016",
+    "module": "esnext",
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
+    "strict": false,
+    "skipLibCheck": true,
```
Comment (mattstrom) [inline tsconfig.json:8]:
```suggestion
    "skipLibCheck": true,
    "lib": [
      "dom"
    ]
```

File: src/types.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+export interface Evolv {
+	client: {
+		getDisplayName: (type: string, id: string) => Promise<string>;
+		on: (event: string, callback: () => void) => void;
+	};
+	context: {
+		get: (key: string) => Experiments;
+		remoteContext: RemoteContext;
+	}
+}
```
Comment (mattstrom) [inline src/types.ts:30]: I don't know much we should care, but we could pull these from one of our packages.

File: src/shared/utils.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+export const waitForElement = async (selector: any, timeout = 5000) => {
+	const startTime = Date.now();
+
+	while (document.querySelector(selector) === null) {
+		if (Date.now() - startTime > timeout) {
+			throw new Error(`Timeout waiting for element with selector: ${selector}`);
+		}
+
+		await new Promise((resolve) => requestAnimationFrame(resolve));
+	}
```
Comment (mattstrom) [inline src/shared/utils.ts]: ~I don't think `while` awaits promises~

File: src/popup.ts [hunk 7]
```diff
@@ -63,7 +57,7 @@ const handleExperimentRowClicks = () => {
   waitForElement('.experiment-title-bar').then(function () {
     let experimentRows = document.querySelectorAll('.experiment-title-bar');
 
-    const clickAction = function (e) {
+    const clickAction = function (e: any) {
```
Comment (mattstrom) [inline src/popup.ts]:
```suggestion
    const clickAction = function (e: MouseEvent) {
```

File: src/background.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+  type: string;
+  stage: Stage;
+  envId: string;
+}
+
+chrome.runtime.onMessage.addListener(({ type, stage, envId }: Request, sender, sendMessage): boolean => {
+  if (type === 'evolv:environmentConfig') {
+    const url: string = stage === Stage.Development
+      ? `participants-newdev.evolvdev.com/v1/${envId}`
+      : `https://participants${stage}.evolv.ai/v1/${envId}`;
```
Comment (mattstrom) [inline src/background.ts]: ❓ 
```suggestion
      ? `https://participants-newdev.evolvdev.com/v1/${envId}`
      : `https://participants${stage}.evolv.ai/v1/${envId}`;
```

File: src/contentScript.ts [hunk 9]
```diff
@@ -71,6 +111,7 @@ chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
                 source: 'evoTools',
                 type: 'evolv:refreshContext'
             }, '*');
+            waitForEvolv();
```
Comment (mattstrom) [inline src/contentScript.ts:118]: Dangling promise

**Review by mattstrom — APPROVED** [review]

File: src/shared/utils.ts (hunk 6 above)
Comment (joshua-barnard-evolv) [reply src/shared/utils.ts]: I went with the recursive approach and it feels like it loads more quickly.

**Pull request merged**